client/extensions/auto-translator/src/widget/translation-dialog/compare-accordion/index.tsx

```diff
@@ -0,0 +1,226 @@
+        <span
+          className="sd-input__label sd-input__label--boxed"
+          id={`compare-body-html-${index}-label`}
+        >
+          {capitalize(gettext("body HTML"))}
+        </span>
+        <div className="sd-input__input-container">
+          <div
+            aria-labelledby={`compare-body-html-${index}-label`}
+            dangerouslySetInnerHTML={{ __html: body_html }}
+          ></div>
+        </div>
+      </div>
+    </>
+  );
+};
+
+export const CompareAccordion = () => {
+  const { gettext } = superdesk.localization;
+  const { values } = useFormikContext<TranslationDialogFormProps>();
+  const dmp = new DiffMatchPatch();
+
+  const [isOpen, setIsOpen] = React.useState<boolean>(false);
+  const [compareLeft, setCompareLeft] = React.useState<
+    TranslationDialogFormProps["writethru"]
+  >(getObjectKeys(values.translations)?.[0] ?? "");
+  const [compareRight, setCompareRight] = React.useState<
+    TranslationDialogFormProps["writethru"]
+  >(getObjectKeys(values.translations)?.[0] ?? "");
+
+  return (
+    <ToggleBox
+      variant="simple"
+      title={capitalize(gettext("Compare"))}
+      margin="none"
+      onOpen={() => {
+        setIsOpen(true);
+      }}
+      onClose={() => {
+        setIsOpen(false);
+      }}
+    >
+      {isOpen && (
+        <Container gap="large" direction="column" className="mx-2">
+          <div style={{ width: "100%" }}>
+            <GridList margin="0">
+              <Select
+                value={compareLeft}
+                onChange={(e) => {
+                  setCompareLeft(e.target.value);
+                }}
+                label={`${capitalize(gettext("writethru"))} 1`}
+              >
+                <option value="" hidden></option>
+                {getObjectKeys(values.translations).map((writethru) => (
+                  <option value={writethru} key={`writethru-${writethru}`}>
+                    {capitalize(writethru)}
+                  </option>
+                ))}
+              </Select>
+              <Select
+                value={compareRight}
+                onChange={(e) => {
+                  setCompareRight(e.target.value);
+                }}
+                label={`${capitalize(gettext("writethru"))} 2`}
+              >
+                <option value="" hidden></option>
+                {getObjectKeys(values.translations).map((writethru) => (
+                  <option value={writethru} key={`writethru-${writethru}`}>
+                    {capitalize(writethru)}
+                  </option>
+                ))}
+              </Select>
+            </GridList>
+          </div>
+          {compareLeft && compareRight && (
+            <div style={{ width: "100%" }}>
+              <div
+                className="sd-grid-list sd-grid-list--large sd-grid-list--gap-small sd-margin--0"
+                style={{ gridTemplateColumns: "repeat(3, 1fr)" }}
+              >
+                {COMPARE_VERIONS.map((writethru, index) => {
```

Comment:
can be moved to a function to increase readability.

Reply:
Do you mean the map? this is just setting up values. I've already made component for the content

Reply:
Yeah I mean like the following, we can have a function.
```typescript
const generateComparisonContent = (
  writethru: "left" | "right" | "diff",
  compareLeft: string,
  compareRight: string,
  values: TranslationDialogFormProps,
  dmp: DiffMatchPatch,
  gettext: Function
) => {
  let headline = "", headline_extended = "", body_html = "";
  let header = writethru === "diff" ? capitalize(gettext("diff")) : `${capitalize(gettext("writethru"))} ${writethru === "left" ? 1 : 2}`;

  if (writethru === "left" || writethru === "right") {
    const selectedVersion = writethru === "left" ? compareLeft : compareRight;
    headline = values.translations[selectedVersion].original.headline;
    headline_extended = values.translations[selectedVersion].original.headline_extended;
    body_html = values.translations[selectedVersion].original.body_html;
  } else if (writethru === "diff") {
    const diffHeadline = dmp.diff_main(values.translations[compareLeft].original.headline, values.translations[compareRight].original.headline);
    const diffHeadline_extended = dmp.diff_main(values.translations[compareLeft].original.headline_extended, values.translations[compareRight].original.headline_extended);
    const diffBody_html = dmp.diff_main(values.translations[compareLeft].original.body_html, values.translations[compareRight].original.body_html);

    headline = dmp.diff_prettyHtml(diffHeadline);
    headline_extended = dmp.diff_prettyHtml(diffHeadline_extended);
    body_html = dmp.diff_prettyHtml(diffBody_html);
  }

  return { header, headline, headline_extended, body_html };
};
``` 

We can use it this way

```typescript
{COMPARE_VERIONS.map((writethru, index) => {
  const { header, headline, headline_extended, body_html } = generateComparisonContent(
    writethru,
    compareLeft,
    compareRight,
    values,
    dmp,
    gettext
  );

  return (
    <Container
      key={`${writethru}-${index}`}
      gap="large"
      direction="column"
    >
      <p className="text-md font-medium self-center m-0">
        {header}
      </p>
      <CompareContent
        headline={headline}
        headline_extended={headline_extended}
        body_html={body_html}
        index={index}
      />
    </Container>
  );
})}
``` 

Just for increasing readability and try to keep dom return statement lean.

Reply:
done